src/RequestExecutor/Request.ts

```diff
@@ -239,10 +240,17 @@ export class Request {
     }
   }
 
-  private matchHostname(hostname: string, wildcard: string): boolean {
-    return (
-      wildcard === hostname || Helpers.wildcardToRegExp(wildcard).test(hostname)
-    );
+  private matchHostnameAndPort(
+    hostname: string,
+    port: string,
+    cert: Cert
+  ): boolean {
+    const hostNameMatch =
+      cert.hostname === hostname ||
+      Helpers.wildcardToRegExp(cert.hostname).test(hostname);
+    const portMatch = cert.port === port;
```

Comment:
Should one consider to test protocols here as well?
Cause both might be empty ones:
1. `cert.port === ''` for `https://domain:443`
1. `port === ''` for `this.url == 'http://domain:80'`

Thus `portMatch` will be true.

But I'm not sure is this real life case for repeater to handle pure http request?

Reply:
Ok, now port is always got some value (80, 443 applied as fallback).
But cert.port could be undefined or empty string. And we have to use same fallback for that. Right?

Reply:
I've checked the code and got the point. Thx.
The sequence of conditions check inside `matchHotsnameAndPort` was a bit unclear, that's why I didn't grasp the idea from the beginning. Therefore I've deciced to sinplify it a bit with early exist logic. Pls have a look: https://github.com/NeuraLegion/bright-cli/pull/662/commits/541e7e8d624db1b9c72d5066edb44afc9b21190a

P.S.: you may rework it further before merging, if it seems required to you.

Reply:
@aborovsky can you please make sure tests are passing now ? It was ok before you commit